src/main/java/core/basesyntax/storage/Storage.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.storage;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+    private final Map<String, Integer> fruits = new HashMap<>();
+
+    public Map<String, Integer> getFruits() {
+        return fruits;
+    }
+
+    public void put(String fruit, Integer quantity) {
+        fruits.put(fruit, quantity);
+    }
+
+    public void plus(String fruit, Integer quantity) {
+        Integer initialAmount = fruits.get(fruit);
+        if (initialAmount == null) {
+            throw new RuntimeException("Can't purchase not existent fruit");
+        }
+        if (quantity > initialAmount) {
+            throw new RuntimeException("Can't remove more than there is in stock");
+        }
+        fruits.put(fruit, initialAmount + quantity);
+    }
```

Comment:
the checks you have here make no sense if you actually adding some stock to the storage, why would you check if it is more or null? purchasing is a minus, not a plus
\+ this is not the responsibility of the Storage, so let's just make it public and move all of the other logic to the handlers